firedrake/variational_solver.py

```diff
@@ -126,12 +130,10 @@ def dm(self):
 class NonlinearVariationalSolver(OptionsManager, NonlinearVariationalSolverMixin):
     r"""Solves a :class:`NonlinearVariationalProblem`."""
 
-    DEFAULT_SNES_PARAMETERS = {"snes_type": "newtonls",
-                               "snes_linesearch_type": "basic"}
+    DEFAULT_SNES_PARAMETERS = DEFAULT_SNES_PARAMETERS
 
     # Looser default tolerance for KSP inside SNES.
-    DEFAULT_KSP_PARAMETERS = solving_utils.DEFAULT_KSP_PARAMETERS.copy()
-    DEFAULT_KSP_PARAMETERS["ksp_rtol"] = 1e-5
+    DEFAULT_KSP_PARAMETERS = MappingProxyType(DEFAULT_KSP_PARAMETERS | {'ksp_rtol': 1e-5})
```

Comment:
Is this | (pipe) syntax compatible with python 3.8?
Sorry if it is not the proper place to comment.
Enrico

Reply:
Err.... no. Good spot! This is the right place to comment, but unfortunately the code is already merged. I will open a new pull request to remove this.

**However**, we will very shortly be dropping support for Python 3.8 as Python will [soon be dropping support](https://devguide.python.org/versions/).

Reply:
Ok, thanks. Actually everything worked fine on 3.9 and 3.10.
Next time, I will try to do the pull request by myself.
Enrico